### ISSUE TITLE
Reconsider baked-in <link> path resolution

### ISSUE THREAD

**Issue by jrunning** [webpack-contrib/polymer-webpack-loader#34]
The loader turns `<link>` `href`s into absolute paths, [here](https://github.com/webpack-contrib/polymer-webpack-loader/blob/2437f125821b8ebac2106771bf982d3939e09f65/src/index.js#L55):

```js
path = osPath.join(osPath.dirname(this.currentFilePath), href);
```

This is contrary to the README's claim:

> The loader transforms your components:
>  - `<link rel="import" href="./my-other-element.html">` -> `import './my-other-element.html';`

The example `<link>` is *actually* transformed into something like the following:

```js
import '/absolute/path/to/component/my-other-element.html';
```

I think the implementation should be changed, not the README. Transforming `href`s into `import`s with absolute paths breaks Webpack module resolution features. For example, I can't do this:

```html
<link rel="import" href="some-other-loader!./my-other-element.html">
```

And instead of being able to do this:

```html
<link rel="import" href="@npm-polymer/paper-button/paper-button.html">
```

...I have to do this:

```html
<link rel="import" href="../../node_modules/@npm-polymer/paper-button/paper-button.html">
```

It also makes it impossible to use [module.alias](https://webpack.js.org/configuration/resolve/#resolve-alias), so I can't do this:

```js
alias: {
  'paper-button': path.resolve(__dirname, 'node_modules/@npm-polymer/paper-button'),
},
```

...which would allow me to do this:

```html
<link rel="import" href="paper-button/paper-button.html">
```

I don't think polymer-webpack-loader should monkey with paths at all. It should do like the README says and do a direct mapping of `<link rel="import" href="%anything%">` to `import '%anything%';`.

If I get a chance I'll work up a PR for this.

**Comment by ChadKillingsworth** [comment]
> do a direct mapping of <link rel="import" href="xyz"> to import 'xyz';

Has to be a bit more than that. That import statement is no longer relative but is attempting to lookup `xyz` from the node_modules folder. At a minimum it would need to be `import './xyz'`.

**Comment by ChadKillingsworth** [comment]
Also, see some of the discussion at https://github.com/webpack-contrib/polymer-webpack-loader/issues/29#issuecomment-316040105 (it's tangentially related).

**Comment by jrunning** [comment]
`<link rel="import" href="xyz">` was a misleading example. I meant `xyz` to stand in for "literally any path Webpack knows how to resolve." And that's the thing: Webpack does module resolution, and we should let it. If you have `'node_modules'` in [resolve.modules](https://webpack.js.org/configuration/resolve/#resolve-modules), then you can do this:

```html
// was <link rel="import" href="some-module/some-element.html">
import 'some-module/some-element.html';
```

...and Webpack will look in `node_modules` and see `some-module` and correctly import `node_modules/some-module/some-element.html` using the loader. I have this working in a branch.

**Comment by mheatleyTMN** [comment]
Just as another datapoint. I understand the difficulty, but speaking as an outsider who just started using polymer-webpack-loader I assumed it would respect the aliases configured in webpack.  I am trying to convert an existing large project to webpack and we have several server-side paths that need to be matched up in webpack.

**Comment by ChadKillingsworth** [comment]
Fixed by #40
